cli/cmd/uninstall/uninstall.go

```diff
@@ -345,6 +374,53 @@ func (c *Command) uninstallHelmRelease(releaseName, namespace, releaseType strin
 		}
 	}
 
+	// Delete any custom resources managed by Consul. If they cannot be deleted,
+	// patch the finalizers to be empty on each one.
+	if releaseType == common.ReleaseTypeConsul {
+		uiLogger("Deleting custom resources managed by Consul")
+		crds, err := c.fetchCustomResourceDefinitions()
+		if err != nil {
+			return fmt.Errorf("unable to fetch Custom Resource Definitions for Consul deployment: %v", err)
+		}
+		kindToResource := mapCRKindToResourceName(crds)
```

Comment:
CR patching and deletion searches for the custom resources by their resource name. But the CR itself only knows its Kind. So I make a map of Kind to resource name using the CRDs which know both and pass it to the patching and deletion functions so they can find the resources.